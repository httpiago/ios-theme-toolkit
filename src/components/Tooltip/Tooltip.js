import React from 'react'
import PropTypes from 'prop-types'
import { mapClassList } from '../../Utils'
import Popover from '../Popover/Popover'

/**
 * Tooltip component.
 *
 * @since 0.2.0
 * @see https://httpiago.github.io/ios-theme-toolkit/#/components/Tooltip
 */
export default function Tooltip({ text, className: aditionalClasses, ...rest }) {

  const prefix = 'tooltip'
  const classes = mapClassList({
    // [`${prefix}--${color}`]: color,
  })

  return (
    <Popover
      trigger="hover"
      backdrop={false}
      color="black"
      content={text}
      className={`${prefix} ${classes} ${aditionalClasses}`}
      {...rest}
    />
  )
}

Tooltip.defaultProps = {
  className: '',
  withoutArrow: false,
  mouseEnterDelay: 600
}

Tooltip.propTypes = {
  text: PropTypes.string.isRequired,
  /** Aditional classes. */
  className: PropTypes.string,
  /** Without arrow */
  withoutArrow: PropTypes.bool,
  /** Delay for show tooltip */
  mouseEnterDelay: PropTypes.number
}
