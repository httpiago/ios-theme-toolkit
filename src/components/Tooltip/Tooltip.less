@import '../../customizations.less';

// Default style
@tooltip-opacity: .72;
.tooltip {
  opacity: @tooltip-opacity;
  animation: tooltip-show 280ms ease;
  pointer-events: none;
}

.tooltip .popover__content {
  padding: 8px 12px;
}

@keyframes tooltip-show {
  from { opacity: 0; }
  to { opacity: @tooltip-opacity; }
}
