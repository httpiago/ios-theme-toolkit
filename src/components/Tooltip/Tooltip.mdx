---
name: Tooltip
route: /components/Tooltip
menu: Components
---

import { Playground, PropsTable } from 'docz'
import { Tooltip } from '../../index.js'
import '../../../dist/styles.css'

# Tooltip

To use this component, simply import it into any file and pass the [modifiers](#/#props) in the properties.

```jsx
import { Tooltip } from 'ios-theme-toolkit'

export default () => (
  <Tooltip text={'Hello world!'}>Hover me</Tooltip>
)
```

## Examples

<Playground>
  <Tooltip text="Hello world!">
    Hover me
  </Tooltip>
  <br/><br/>

  <Tooltip withoutArrow text="Hello world!">
    Without arrow
  </Tooltip>
  <br/><br/>

  <Tooltip placement="right" text="Hello world!">
    Right
  </Tooltip>
</Playground>

## Props

<PropsTable of={ Tooltip } />

All the properties which are not listed above will be transferred to the \<div class="tooltip"\> tag.
