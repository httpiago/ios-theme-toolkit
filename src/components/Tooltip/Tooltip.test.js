/**
 * TEST FILE
 * Please try writing the tests for all component behaviors,
 * manually call the functions within the class, change the properties...
 *
 * @see https://jestjs.io/docs/en/expect
 * @see https://github.com/airbnb/enzyme/tree/master/docs/api
 */
import React from 'react'
import { shallow } from 'enzyme'
import Tooltip from './Tooltip'

const COMPONENT = shallow(
  <Tooltip text={'Hello world!'}>
    <div className="check-children">Hover me</div>
  </Tooltip>
)

describe('Test Tooltip component', () => {
  it('should render correctly based on last snapshot', () => {
    expect(COMPONENT).toMatchSnapshot()
  })

})
