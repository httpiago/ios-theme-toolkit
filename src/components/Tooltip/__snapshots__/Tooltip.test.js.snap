// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tooltip component should render correctly based on last snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tooltip
    className=""
    mouseEnterDelay={600}
    text="Hello world!"
    withoutArrow={false}
  >
    <div
      className="check-children"
    >
      Hover me
    </div>
  </Tooltip>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "backdrop": false,
      "children": <div
        className="check-children"
      >
        Hover me
      </div>,
      "className": "tooltip  ",
      "closeOnInnerClick": false,
      "color": "black",
      "content": "Hello world!",
      "mouseEnterDelay": 600,
      "placement": "bottom",
      "trigger": "hover",
      "withoutArrow": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hover me",
        "className": "check-children",
      },
      "ref": null,
      "rendered": "Hover me",
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "backdrop": false,
        "children": <div
          className="check-children"
        >
          Hover me
        </div>,
        "className": "tooltip  ",
        "closeOnInnerClick": false,
        "color": "black",
        "content": "Hello world!",
        "mouseEnterDelay": 600,
        "placement": "bottom",
        "trigger": "hover",
        "withoutArrow": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hover me",
          "className": "check-children",
        },
        "ref": null,
        "rendered": "Hover me",
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
